dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,488 @@
+    {'loss': 'log', 'average': False, 'alpha': 0.0080502}
+
+    Attributes
+    ----------
+    cv_results_ : dict of lists
+        Information about the cross validation scores for each model.
+        All lists are ordered the same, and this value can be imported into
+        a pandas DataFrame. This dict has keys of
+
+        * ``rank_test_score``
+        * ``model_id``
+        * ``mean_fit_time``
+        * ``mean_score_time``
+        * ``std_fit_time``
+        * ``std_score_time``
+        * ``mean_test_score``
+        * ``test_score``
+        * ``partial_fit_calls``
+        * ``params``
+
+    metadata_ : dict
+        Information about every model that was trained. This variable can also
+        be obtained without fitting through
+        :func:`~dask_ml.model_selection.HyperbandCV.metadata`.
+    history_ : list of dicts
+        Information about every model after every time it is scored.
+    best_params_ : dict
+        The params that produced the best performing model
+    best_estimator_ : any
+        The best performing model
+    best_index_ : int
+        The index of the best performing model to be used in ``cv_results_``.
+    n_splits_ : int
+        The number of cross-validation splits.
+    best_score_ : float
+        The best validation score on the test set.
+    best_params_ : dict
+        The params that are given to the model that achieves ``best_score_``.
+
+    Notes
+    -----
+
+    Hyperband is an adaptive model selection scheme that spends time on
+    high-performing models, because our goal is to find the highest performing
+    model. This means that it stops training models that perform poorly.
+
+    Setting Hyperband parameters
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    To set ``max_iter`` and the chunk size for ``X`` and ``y``, you need to
+    know
+
+    * how many "epochs" or "passes through ``X``" to train the model for
+      (``epochs`` below)
+    * how many parameters to sample (``params_to_sample`` below)
+
+    To determine the chunk size and ``max_iter``,
+
+    1. Set ``frac = epochs / params_to_sample``, where ``frac`` is ``chunks / len(X)``
+    2. Set ``max_iter = params_to_sample``
+
+    The number of parameters to sample will depend on how complex the search
+    space is. If you're tuning many parameters, you'll need to increase
+    ``params_to_sample``.
+
+    Limitations
+    ^^^^^^^^^^^
+    There are some limitations to the `current` implementation of Hyperband:
+
+    1. The full dataset is requested to be in distributed memory
+    2. The testing dataset must fit in the memory of a single worker
+
+    References
+    ----------
+    .. [1] "Hyperband: A novel bandit-based approach to hyperparameter
+           optimization", 2016 by L. Li, K. Jamieson, G. DeSalvo, A.
+           Rostamizadeh, and A. Talwalkar.  https://arxiv.org/abs/1603.06560
+
+    """
+
+    def __init__(
+        self,
+        model,
+        params,
+        max_iter,
```

Comment:
This should probably have a default, even if it isn't ideal. Can you turn the logic in the doc note into a heuristic for choosing a default `max_iter`?

Reply:
I'll need to think about a good default value, and how it relates to the logic I have.

The corresponding parameter in Scikit-Learn is `n_iter` in [RandomizedSearchCV][1], which defaults to 10 parameters to sample.


[1]:http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html#sklearn.model_selection.RandomizedSearchCV

Reply:
I chose the default to be `max_iter=81`. I think that's the lowest it can be to realize moderate adaptive gains.